smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -347,12 +444,35 @@ contract BridgeBank is BankStorage,
         return decimals;
     }
 
-    /*
-     * @dev: Locks received Ethereum/ERC20 funds.
-     *
-     * @param recipient: bytes representation of destination address.
-     * @param token: token address in origin chain (0x0 if ethereum)
-     * @param amount: value of deposit
+    /**
+     * @dev Fetches the denom of a token by address
+     * @param token The bridgeTokens's address
+     * @return The bridgeTokens's denom or ''
+     */
+    function getDenom(address token) private returns (string memory) {
+        string memory denom = contractDenom[token];
```

Comment:
Would be great to see if we could pack all the things, cosmos denom, token name, symbol and decimals into a single struct for gas savings.

Reply:
Ok, I've just tested it. In this specific case, we can't save gas by packing these variables because of their types.
We have 3 `string` and one `uint8`, so we'll always use 4 storage slots no matter what. Gas costs stay the same in most cases and operations end up being more expensive whenever we only need to query/set fewer variables.
I'm dropping this one unless you feel we should discuss it further, ok?

Reply:
As discussed via Slack:
"Sounds good, we don’t need to move to using a struct"